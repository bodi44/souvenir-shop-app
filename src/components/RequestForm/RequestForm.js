import React from 'react'
import PropTypes from 'prop-types'

import './RequestForm.scss'
import BEM from '../../helpers/BEM'

const b = BEM('RequestForm')

const RequestForm = ({ buttonStyle, handleInputChange, handleSubmit, handleButtonColorChange }) => (
  <form action="" className={b()} onSubmit={e => handleSubmit(e)}>
    <input
      type="text"
      name='query'
      placeholder="Type your query here..."
      className={b('input')}
      onChange={handleInputChange}
    />
    <button
      className={b('submit')}
      type='submit'
      style={{ backgroundColor: buttonStyle }}
      onMouseDown={handleButtonColorChange}
      onMouseUp={handleButtonColorChange}>
      Click to Submit
    </button>
  </form>
)

RequestForm.propTypes = {
  buttonStyle: PropTypes.string,
  handleInputChange: PropTypes.func,
  handleSubmit: PropTypes.func,
  handleButtonColorChange: PropTypes.func
}

export default RequestForm