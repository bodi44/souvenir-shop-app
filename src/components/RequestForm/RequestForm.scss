.RequestForm {
  display: flex;
  flex-direction: row;
  align-items: center;
  margin: 0 auto;
  width: 100%;

  &__input {
    font-family: 'Roboto', sans-serif;
    margin-right: -4px;
    align-self: center;
    padding: 10px;
    border-radius: 5px;
    border: 2px solid #bdc3c7;
    width: 100%;
  }

  &__submit {
    font-size: 16px;
    font-family: 'Roboto', sans-serif;
    border: none;
    background-color: #bdc3c7;
    border-radius: 0 5px 5px 0 ;
    width: 40%;
    padding: 10px;
    font-weight: 600;
    white-space: nowrap;

    &:hover{
      background-color: #7f8fa6;
    }
  }
}