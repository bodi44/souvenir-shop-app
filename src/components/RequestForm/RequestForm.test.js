import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import expect from 'expect'
import Adapter from 'enzyme-adapter-react-16'

import RequestForm from './RequestForm'
import ProviderWrapper from '../Provider'
import configureStore from '../../configureStore'

Enzyme.configure({ adapter: new Adapter() })

describe('RequestForm component', () => {
  it('should render correctly with given props', () => {
    const allPurchases = [
      {
        id: '2019-04-25',
        purchases: [
          {
            id: 'c4f63a46-de46-49c3-9773-0a47c62cdb77',
            date: '2019-04-25',
            price: 12,
            currency: 'EUR',
            productName: 'Photo'
          }
        ]
      }
    ]

    const currencyData = {
      'BTC': 0.000211,
      'CZK': 25.640122,
      'EUR': 1,
      'GBP': 0.860335,
      'UAH': 29.569446,
      'USD': 1.120097
    }

    const component = shallow(
      <ProviderWrapper store={configureStore()}>
        <RequestForm allPurchases={allPurchases} currencyData={currencyData}/>
      </ProviderWrapper>)
    expect(component).toMatchSnapshot()
  })
})