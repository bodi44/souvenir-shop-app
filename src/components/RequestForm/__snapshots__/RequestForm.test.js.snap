// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestForm component should render correctly with given props 1`] = `ShallowWrapper {}`;
